### PR TITLE
CIV-9616 Defendant Latest Update Mediation Succesful PDF

### PR DESCRIPTION
**Pull request by jeswanth-hmcts**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/CIV-9616

### Change description ###

Defendant Latest Update Mediation Succesful PDF

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MediationSuccessfulCallbackHandler.java [hunk 1]
```diff
@@ -41,12 +49,47 @@ public List<CaseEvent> handledEvents() {
...
             .data(caseDataUpdated.toMap(objectMapper))
             .state(CASE_STAYED.name())
             .build();
     }
+
+    private List<Element<ManageDocument>> updateManageDocumentsListWithMediationAgreementDocument(
+        CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        List<Element<ManageDocument>> manageDocumentsList =
+            Optional.ofNullable(caseData.getManageDocuments()).orElse(new ArrayList<>());
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MediationSuccessfulCallbackHandler.java:70]: we are repeating code here and in the further up. perhaps this knowledge should be inside casedata. then we will just have to call that method on case data here and elsewhere where we would require it. In addition we could unit test this method on case data test

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MediationSuccessfulCallbackHandler.java [hunk 2]
```diff
@@ -41,12 +49,47 @@ public List<CaseEvent> handledEvents() {
...
+                .documentLink(document.getDocument())
+                .documentName(document.getName())
+                .documentType(mapDocumentTypeToManageDocumentType(document.getDocumentType()))
+                .build();
+            manageDocumentsList.add(element(manageDocument));
+        });
+        return manageDocumentsList;
+    }
+
+    private ManageDocumentType mapDocumentTypeToManageDocumentType(DocumentType documentType) {
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MediationSuccessfulCallbackHandler.java:87]: this mapping should not be inside the handler. what if this knowledge is needed else where? hanlder should not have a lot of business logic. think about it as a controller. perhaps the mapping should happen in the ManageDocumentType enum?

Comment (jeswanth-hmcts) [reply src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MediationSuccessfulCallbackHandler.java]: moved to a util class

File: src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java [hunk 3]
```diff
@@ -982,13 +982,19 @@ public List<Element<RecurringIncomeLRspec>> getRecurringIncomeForRespondent1() {
...
+                .orElse(
+                    null);
         }
         return Optional.ofNullable(getRespondent1DQ()).map(Respondent1DQ::getRespondent1DQRecurringExpenses).orElse(
             null);
     }
 
+    @JsonIgnore
+    public List<Element<ManageDocument>> getManageDocumentsList() {
+        return Optional.ofNullable(getManageDocuments()).orElse(new ArrayList<>());
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java:995]: Collections.emptyList() would be better

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MediationSuccessfulCallbackHandler.java [hunk 4]
```diff
@@ -41,12 +47,37 @@ public List<CaseEvent> handledEvents() {
...
             .data(caseDataUpdated.toMap(objectMapper))
             .state(CASE_STAYED.name())
             .build();
     }
+
+    private List<Element<ManageDocument>> updateManageDocumentsListWithMediationAgreementDocument(
+        CallbackParams callbackParams) {
+        CaseData caseData = callbackParams.getCaseData();
+        List<Element<ManageDocument>> manageDocumentsList = caseData.getManageDocumentsList();
+        Optional<MediationAgreementDocument> mediationAgreementDocument = Optional.ofNullable(caseData)
```
Comment (TatsyMaslova) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/MediationSuccessfulCallbackHandler.java:68]: how come caseData could be null? perhaps it should be Optional.ofNullable(casdata.getMediation()).map(Mediation::getMediationSuccessful).map(MediationSuccssful::getMediationAgreement)

**Review by laperla11 — APPROVED** [review]

**Review by TatsyMaslova — APPROVED** [review]

**Review by ElifDenizEsea — APPROVED** [review]

**Review by neeta-hmcts — APPROVED** [review]

**Review by rajakm — APPROVED** [review]

**Pull request merged**